plugins/modules/network/netscaler/netscaler_nitro_request.py

```diff
@@ -642,10 +642,11 @@ def get_filtered(self):
         if self._module.params['filter'] is None:
             self.fail_module(msg='NITRO filter is undefined.')
 
-        keys = list(self._module.params['filter'].keys())
-        filter_key = keys[0]
-        filter_value = self._module.params['filter'][filter_key]
-        filter_str = '%s:%s' % (filter_key, filter_value)
+        filter_str = ''
+        for key, value in self._module.params['filter'].items():
+            if len(filter_str) != 0:
```

Comment:
The more Pythonic way for this would be:
```suggestion
            if filter_str:
```

Reply:
(In fact, the whole thing could be replaced by `filter_str = '&'.join('%s, %s' % (k, v) for k, v in self._module.params['filter'].items())'. With of course `'&'` replaced by the correct string, probably `'%26'`.)

Reply:
That's idd an even better solution, will change. Though see my other comment, that `&` doesn't need to be encoded.